server/src/main/java/io/crate/exceptions/SQLExceptions.java

```diff
@@ -111,81 +106,28 @@ public static boolean isShardFailure(Throwable e) {
     }
 
     public static Function<Throwable, Exception> forWireTransmission(AccessControl accessControl) {
-        return e -> createSQLActionException(e, accessControl::ensureMaySee);
+        return e -> handleException(e, accessControl::ensureMaySee);
     }
 
-    public static SQLActionException forWireTransmission(AccessControl accessControl, Throwable e) {
-        return createSQLActionException(e, accessControl::ensureMaySee);
+    public static RuntimeException forWireTransmission(AccessControl accessControl, Throwable e) {
+        return handleException(e, accessControl::ensureMaySee);
     }
 
-    /**
-     * Create a {@link SQLActionException} out of a {@link Throwable}.
-     * If concrete {@link ElasticsearchException} is found, first transform it
-     * to a {@link CrateException}
-     */
-    public static SQLActionException createSQLActionException(Throwable e, Consumer<Throwable> maskSensitiveInformation) {
-        // ideally this method would be a static factory method in SQLActionException,
-        // but that would pull too many dependencies for the client
-
-        if (e instanceof SQLActionException) {
-            return (SQLActionException) e;
-        }
+    public static RuntimeException handleException(Throwable e, @Nullable Consumer<Throwable> maskSensitiveInformation) {
```

Comment:
`handle` is a bit generic and there is no hint how it is handling the exception. Maybe this could also be called `forWireTransmission` ? Or we rename all 3 methods to `prepareForClientTransmission` or something like that?